torch/distributions/transforms.py

```diff
@@ -357,7 +357,7 @@ def _inverse(self, y):
         return y.log() - (-y).log1p()
 
     def log_abs_det_jacobian(self, x, y):
-        return -(y.reciprocal() + (1 - y).reciprocal()).log()
+        return -torch.log(1e-6 + y.reciprocal() + (1 - y).reciprocal())
```

Comment:
This will not address the case when `y=1.`, where we will still see infinity. How about we clamp this using 

```python
clamped = torch.clamp(y.reciprocal() + (1 - y).reciprocal(), max=torch.finfo(y.dtype).max)
return -clamped.log()
```

Reply:
It looks good ! I've tried a testing, it seems when std from Normal distribution is large, there is still possibliity to get `'inf`, e.g.

```python
d = TransformedDistribution(Normal(torch.tensor(0.0), torch.tensor(100.0)), [StableSigmoidTransform()])
x = d.sample()
print(x)
d.log_prob(x)
>>> tensor(1.)
>>> tensor(-inf)
```

Reply:
Yes, that would just fix the jacobian term, but in `TransformedDistribution.log_prob` we still evaluate `base_dist.log_prob(y)` and in this case `y = inf`. I think what you really want is a something like a `ClippedSigmoidTransform` which differs from `SigmoidTransform` in that it clips the output of `_call` to lie in (0, 1):

```python
    def _call(self, x):
        finfo = torch.finfo(x.dtype)
        clamped = torch.clamp(torch.sigmoid(x), min=finfo.tiny, max=1. - finfo.eps)
        return clamped
```

Then you shouldn't see any `inf`s in your example above.

Reply:
@alicanb, @fritzo - Do you think this is something we should do by default (tensorflow doesn't seem to do this either), or have a separate transform class `ClippedSigmoidTransform`?

Reply:
@neerajprad Thanks a lot for the proposal. I think this is rather important for some applications where we evaluate `log_prob` of a sample drawn from a transformed distribution that is conditional on the input of the NN. Without such fixes, it is easily happen to get NaN/Inf. An example use case is tanh transformed policy network in reinforcement learning context, e.g. SAC algorithm.

Reply:
Or this can be provided with an optionally flag in `SigmoidTransform`, without creating a new class.

Reply:
I think that’s a good idea - we can have an optional keyword arg, clip=False.

Reply:
Sorry, I'm late to the party. @neerajprad your suggestion sounds great, do you think users   would want to change clip limits (@zuoxingdong, have you ever had the need?) ? If that's the case, instead of a bool arg, we can keep the clamping symmetric (min=lim, max=1-lim) and have lim as the argument. If not, then bool arg sound great